hamocc/bodensed.F90

```diff
@@ -79,33 +82,60 @@ subroutine bodensed(kpie,kpje,kpke,pddpo)
      write(io_stdo_bgc,*)  ' '
   endif
 
-  porwat(1) = 0.85
-  porwat(2) = 0.83
-  porwat(3) = 0.8
-  porwat(4) = 0.79
-  porwat(5) = 0.77
-  porwat(6) = 0.75
-  porwat(7) = 0.73
-  porwat(8) = 0.7
-  porwat(9) = 0.68
-  porwat(10) = 0.66
-  porwat(11) = 0.64
-  porwat(12) = 0.62
+  ! this initialization can be done later via reading a porosity map
+  if (l_sed_por)then
+   ! lon-lat variable sediment porosity from input file
+   call apply_sedpor(kpie,kpje,ks,omask,sed_por)
+  else
+   porwat(:,:,1) = 0.85
+   porwat(:,:,2) = 0.83
+   porwat(:,:,3) = 0.8
+   porwat(:,:,4) = 0.79
+   porwat(:,:,5) = 0.77
+   porwat(:,:,6) = 0.75
+   porwat(:,:,7) = 0.73
+   porwat(:,:,8) = 0.7
+   porwat(:,:,9) = 0.68
+   porwat(:,:,10) = 0.66
+   porwat(:,:,11) = 0.64
+   porwat(:,:,12) = 0.62
+  endif
 
   if (mnproc == 1) then
-     write(io_stdo_bgc,*)  'Pore water in sediment: ',porwat
+     write(io_stdo_bgc,*)  'Pore water in sediment initialized'
   endif
 
   seddzi(1) = 500.
   do k = 1, ks
-     porsol(k) = 1. - porwat(k)
-     if(k >= 2) porwah(k) = 0.5 * (porwat(k) + porwat(k-1))
-     if(k == 1) porwah(k) = 0.5 * (1. + porwat(1))
      seddzi(k+1) = 1. / dzs(k+1)
      seddw(k) = 0.5 * (dzs(k) + dzs(k+1))
+     do j = 1, kpje
+     do i = 1, kpie
+        porsol(i,j,k) = 1. - porwat(i,j,k)
+        if(k >= 2) porwah(i,j,k) = 0.5 * (porwat(i,j,k) + porwat(i,j,k-1))
+        if(k == 1) porwah(i,j,k) = 0.5 * (1. + porwat(i,j,1))
+     enddo
+     enddo
   enddo
+  
+  sedict = 1.e-9 * dtbgc ! Moecular diffusion coefficient
+  ! Dissolution rate constant of opal (disso) [1/(kmol Si(OH)4/m3)*1/sec]
+  ! THIS NEEDS TO BE CHANGED TO disso=3.e-8! THIS IS ONLY KEPT FOR THE MOMENT
+  ! FOR BACKWARDS COMPATIBILITY
+  !disso=3.e-8  ! (2011-01-04) EMR
+  !disso=1.e-6 ! test vom 03.03.04 half live sil ca. 20.000 yr 
+  disso_sil = 1.e-6*dtbgc
+  ! Silicate saturation concentration is 1 mol/m3
+  silsat    = 0.001
+
+  ! Degradation rate constant of POP (disso) [1/(kmol O2/m3)*1/sec]
+  disso_poc = 0.01 / 86400. * dtbgc  !  disso=3.e-5 was quite high
```

Comment:
```suggestion
  disso_poc = 0.01 / 86400. * dtbgc
```
Is this comment relevant, do we need to keep it around?

Reply:
I can't tell - maybe @JorgSchwinger made some experiences that allow for better judgement?